common/utilities/timing.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import time
+from functools import wraps
+
+
+class StopWatch:
+
+    start_time = None
+
+    def start_timer(self):
+        self.start_time = time.time()
+
+    def stop_timer(self):
+        end_time = time.time()
+        diff = end_time - self.start_time
+        return diff
+
+
+def time_method(method):
+    """
+    A method decorator that logs the time taken to execute a given method
+
+    :param method:
+    :return:
+    """
+
+    @wraps(method)
+    def invoke_method_with_timer(*args, **kwargs):
+        stopwatch = StopWatch()
+        stopwatch.start_timer()
+        try:
+            return method(*args, **kwargs)
+        finally:
+            duration = stopwatch.stop_timer()
+            duration = round(duration, 3)
+            logging.info({
+                'method': method.__name__,
+                'duration': duration
+            })
+
+    return invoke_method_with_timer
+
+
+def time_method_async(method):
+    """
+    A method decorator that logs the time taken to execute a given method. This decorator should be used to wrap
+    asynchronous methods as the standard time_method decorator does not handle async methods effectively
+    """
+
+    @wraps(method)
+    async def invoke_method_with_timer(*args, **kwargs):
+        stopwatch = StopWatch()
+        stopwatch.start_timer()
+        try:
+            return await method(*args, **kwargs)
+        finally:
+            duration = stopwatch.stop_timer()
+            duration = round(duration, 3)
+            logging.info({
+                'method': method.__name__,
+                'duration': duration
+            })
+
+    return invoke_method_with_timer
+
+
+def log_request(method):
```

Comment:
Probably worth naming this method (and `log_request_async`) to reflect the fact that we're doing timing. Perhaps `time_request`/`time_request_async`, for consistency with the other methods provided?

Reply:
👍